### ISSUE TITLE
Ganeti.Jobs' waitForJobs breaks after waiting for ~10 minutes, breaking some hbal runs

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#773]
Originally reported of Google Code with ID 717.

```
What software version are you running? Please provide the output of "gnt-cluster --version", "gnt-cluster version", and "hspace --version".

# gnt-cluster --version; gnt-cluster version; hspace --version
gnt-cluster (ganeti v2.10.0rc2) 2.10.0~rc2
Software version: 2.10.0~rc2
Internode protocol: 2100000
Configuration format: 2100000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.10.0rc2
hspace (ganeti) version v2.10.0rc2
compiled with ghc 6.12
running on linux x86_64

<b>What distribution are you using?</b>

Debian Squeeze.

<b>What steps will reproduce the problem?</b>

1. Run hbal -L -X
2. Wait until one of the jobs that hbal submits takes more than ~10 minutes to complete
3. hbal sleeps forever ("select(0, [], [], NULL, {17592186044416, 20000}" in strace)

<b>What is the expected output? What do you see instead?</b>

Expected: hbal does not try to outlive us all.

<b>Please provide any additional information below.</b>

waitForJobs in src/Ganeti/Jobs.hs has an integer overflow that (at least on amd64) causes it to break after waiting for ~10 minutes. This results in hbal sleeping forever (when compiled with squeeze's ghc 6.12.1) or crashing (when compiled with precise's ghc 7.4.1) whenever it has to wait ~10 minutes for a jobset to complete.

It appears the bug was introduced in b8e76da. I've attached a suggested fix that remedies the problem.
```
Originally added on 2014-02-13 12:02:36 +0000 UTC.

Attachments: 
[waitforjobs-overflow-fix.patch](https://storage.googleapis.com/google-code-attachments/ganeti/issue-717/comment-0/waitforjobs-overflow-fix.patch)

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-02-13 12:05:24 +0000 UTC.

Added Labels: Priority-Critical
Changed State: Accepted
Added to Milestone: Release2.10

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-02-13 12:25:18 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed in 2.8

commit 95ee8abb01ffdee88f892e6ddc1e23427082e6d6
Author: Michele Tartara <mtartara@google.com>
Date:   Thu Feb 13 13:39:38 2014 +0100

    Fix integer overflow problem in hbal
    
    waitForJobs in src/Ganeti/Jobs.hs has an integer overflow that (at least on
    amd64) causes it to break after waiting for ~10 minutes. This results in hbal
    sleeping forever (when compiled with squeeze's ghc 6.12.1) or crashing (when
    compiled with precise's ghc 7.4.1) whenever it has to wait ~10 minutes for a
    jobset to complete.
    
    Thanks to gjo@google.com for finding the bug and suggesting the fix.
    
    Fixes Issue 717.
    
    Signed-off-by: Michele Tartara <mtartara@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

commit 0e2a45ddf7ac4708a88a75e43102939ee9b61bb0
Author: Michele Tartara <mtartara@google.com>
Date:   Thu Feb 13 12:23:36 2014 +0000

    Add missing space
    
    Also, refactor the line to keep it under 80 chars.
    
    Signed-off-by: Michele Tartara <mtartara@google.com>
    Reviewed-by: Klaus Aehlig <aehlig@google.com>

```
Originally added on 2014-02-13 13:53:51 +0000 UTC.

Mentioned Issue with GoogleCode ID 717 has been migrated to GitHub with ID #773.
Changed State: Fixed
Added to Milestone: Release2.8
Removed from Milestone: Release2.10

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released in the 2.9.5 released.
```
Originally added on 2014-02-27 11:58:33 +0000 UTC.

Changed State: Released
